core/embed/rust/src/ui/component/swipe_detect.rs

```diff
@@ -0,0 +1,370 @@
+                    }
+
+                    return res;
+                }
+            }
+            (Event::Touch(TouchEvent::TouchEnd(pos)), Some(origin)) => {
+                if self.final_animation.is_none() {
+                    // Touch interaction is over, reset the position.
+                    self.origin.take();
+
+                    // Compare the touch distance with our allowed directions and determine if it
+                    // constitutes a valid swipe.
+                    let ofs = pos - origin;
+
+                    let final_value = match self.locked {
+                        // advance in locked direction only trigger animation towards ending
+                        // position
+                        Some(locked) if config.progress(locked, ofs, self.min_trigger()) > 0 => {
+                            Self::PROGRESS_MAX
+                        }
+                        // advance in direction other than locked trigger animation towards starting
+                        // position
+                        Some(_) => 0,
+                        None => {
+                            let mut res = 0;
+                            for dir in SwipeDirection::iter() {
+                                // insta-lock if the movement went at least the trigger distance
+                                if config.progress(dir, ofs, self.min_trigger()) > 0 {
+                                    self.locked = Some(dir);
+                                    res = Self::PROGRESS_MAX;
+                                }
+                            }
+
+                            res
+                        }
+                    };
+
+                    let Some(locked) = self.locked else {
+                        // Touch ended without triggering a swipe.
+                        return None;
+                    };
+
+                    ctx.request_anim_frame();
+                    ctx.request_paint();
+
+                    if !animation_disabled() {
+                        let done = self.moved as f32 / Self::PROGRESS_MAX as f32;
+                        let ratio = 1.0 - done;
+
+                        let duration = config
+                            .duration(locked)
+                            .unwrap_or(Duration::from_millis(Self::DURATION_MS));
+
+                        let duration = ((duration.to_millis() as f32 * ratio) as u32).max(0);
+                        self.final_animation = Some(Animation::new(
+                            self.moved as i16,
+                            final_value,
+                            Duration::from_millis(duration),
+                            Instant::now(),
+                        ));
+                    } else {
+                        // clear animation
+                        self.final_animation = None;
+                        self.moved = 0;
+                        self.locked = None;
+                        return Some(SwipeDetectMsg::Trigger(locked));
```

Comment:
I think this is a bug now? When doing the refactor I didn't understand that you can animate back without triggering.

Now when animations are disabled, and you go back (don't finish the locked swipe), you will still trigger. We'd need to do `if final_value > 0` here, right?
(question is, do we even care?)